### DISCUSSION TITLE
Possiblity of implementing a simple SQLite server

### DISCUSSION THREAD

**Discussion by AdamJSoftware** [databendlabs/openraft#827] [Q&A]
I've noticed the implementation of rocksdb, which is already rather impressive. I've spent the last few weeks trying to find a way to implement a stupid simple distributed SQLite server and there seems to be some hope here with this. So my question is, with the current state of openraft, is implementing SQLite possible? Are there any motives to implement such as with the example of rocksdb? Would we have to recreate SQLite or could we just use the basic sqlite crate?

**Comment by drmingdrmer** [comment]
Openraft can utilize `SqlLite` as a storage backend, primarily for the state-machine, but a separate storage is still required for storing raft logs. 

In this scenario, when `Raft::client_write()` receives a SQL command, Openraft replicates and commits it, applies it to its local state-machine(a local SqlLite instance), and eventually returns the result to the client.

There won't be significant variations between using SqlLite or rocksdb. If the storage implementation differs from the currently provided examples and is a typical use case, I will consider adding another example.
